Add pyqgis cookbook code about selecting a feature with the maptool S… (#6491)
qgis/QGIS-Documentation: docs/pyqgis_developer_cookbook/canvas.rst

> .. index:: Map canvas; Custom map tools
> 

Reviewer: this line refers to the next section then should probably be closer (moved back)
```suggestion
```

> Selection of a feature using QgsMapToolIdentifyFeature

Reviewer: Does it **select**?

------------------------------------------------------

> You can use the map tool QgsMapToolIdentifyFeature for asking to the
> user to select a feature that wiil be send to a callback function.

Reviewer: ```suggestion
You can use the map tool :class:`QgsMapToolIdentifyFeature <qgis.gui.QgsMapToolIdentifyFeature>`
for asking to the user to select a feature that will be sent to a callback function.
```
the :class: tag turns this into a hyperlink pointing to the corresponding section in the pyqgis doc.
I'm not sure I understand the end of the sentence.

.. testcode:: canvas

  def callback(feature):
    """Code called when the feature is selected by the user"""

>     print(f"You clicked on feature {feature.id()}")

Reviewer: ```suggestion
    print("You clicked on feature {}".format(feature.id()))
```
?

> 

Reviewer 2: I'll add the creation of the `canvas` instance:

canvas = iface.mapCanvas()

  feature_identifier = QgsMapToolIdentifyFeature(canvas)

>   # indicates the layer on which the selection wiil be done

Reviewer: ```suggestion
  # indicates the layer on which the selection will be done
```

  feature_identifier.setLayer(vlayer)

>   # use the callback as a slot triggered when the user identify a feature

Reviewer: ```suggestion
  # use the callback as a slot triggered when the user identifies a feature
```

  feature_identifier.featureIdentified.connect(callback)

>   # activation of hte map tool

Reviewer: ```suggestion
  # activation of the map tool
```

  canvas.setMapTool(feature_identifier)

> 
> Writing Custom Map Tools

Reviewer: ```suggestion
.. index:: Map canvas; Custom map tools

Writing Custom Map Tools
```
